openlibrary/templates/lib/nav_head.html

```diff
@@ -1,26 +1,26 @@
 $def with (page)
 
 <header id="header-bar" class="header-bar">
-  <div class="hamburger-component checkbox-menu">
+  <div class="hamburger-component header-dropdown">
```

Comment:
I think it would be great to use a template partial here or make use of `def` so that we keep the HTML markup as consistent as possible across all these components. It's really difficult to see the difference between them all when they are all in the same nav_head.html template.

Reply:
So do we want to break the different components and render them in nav_head.html?

Reply:
Yeh, as far as I can see we have more or less identical HTML, that requires JS and CSS. To aid us in embracing Vue.js in future, I think we need to start thinking in terms of component = template. That will make life easier when we move this to Vue. The nav_head.html template is the header-bar component and that makes use of several of these dropdown components.

Obviously the CSS will vary somewhat - for that we should use parameters. `e.g. header-dropdown('hamburger', , $items);`  and `header-dropdown('more', $items)`. 

We can also take this opportunity to make CSS consistent. Perhaps the hambuger-component class should be renamed hamburger-menu, and account-component should be account-menu to be consistent with more-menu


Does this all make sense?

Reply:
@jdlrobson I just had my 1:1 w/ @bpmcneilly and getting this fix merged in, in some form, is preventing us from speaking w/ accessibility-needed members of our community because the first snags they're inevitably going to hit are with the main header / nav.

If possible, is there an incremental version (such as what's above) that we can land this week and then (@bpmcneilly + @jdlrobson) create a followup issues for the next step of turning this accessibility fix into a componetized solution which meets our code quality & maintainability requirements?

I think Brian is working on that issue now and can update here.

Reply:
I will fix the merge conflict.

Reply:
I will fix the merge conflict.

Reply:
My hope was introducing a template partial would simplify code here and make code review easier as it might tease out some problems with the implementation. If we are running accessibility testing, it would be good to run that on the end 
product rather than an incremental version, and it will likely be easier to address problems with a centralized definition as they arise.

@mekarpeles what deadline are we working towards?

Reply:
@mekarpeles I have a local change that converts the changes brought about in this patch set to template partials. It simplifies some of the CSS and will allow me to do some experimentation with Vue templating this week.

Happy to merge this patch on the basis that my follow-up will get merged within a 2 week period. Does this work for you and your deadline?

Reply:
Yes, this works for me assuming @cdrini is also on board with this direction.

Reply:
Here's a rough patch of how this will look: https://gist.github.com/jdlrobson/f62344bf76d0b82d86852246ed9dfd64

a follow up will port that to Vue.